core/src/main/java/org/apache/accumulo/core/clientImpl/ClientThreadPools.java

```diff
@@ -0,0 +1,210 @@
+      case BULK_IMPORT_POOL:
+        Objects.requireNonNull(config.getNumThreads().get(), "Number of threads must be set");
+        return ThreadPools.createFixedThreadPool(config.getNumThreads().get(), "BulkImportThread",
+            false);
+      case ACTIVE_EXTERNAL_COMPACTION_POOL:
+        Objects.requireNonNull(config.getNumThreads().get(), "Number of threads must be set");
+        return ThreadPools.createFixedThreadPool(config.getNumThreads().get(),
+            "getactivecompactions", false);
+      case SCANNER_READ_AHEAD_POOL:
+        return ThreadPools.createThreadPool(0, Integer.MAX_VALUE, 3L, TimeUnit.SECONDS,
+            "Accumulo scanner read ahead thread", new SynchronousQueue<>(), OptionalInt.empty(),
+            false);
+      case ADD_SPLITS_THREAD_POOL:
+        return ThreadPools.createFixedThreadPool(16, "addSplits", false);
+      case BATCH_SCANNER_READ_AHEAD_POOL:
+        Objects.requireNonNull(config.getNumThreads().get(), "Number of threads must be set");
+        Objects.requireNonNull(config.getThreadName().get(), "Name of threads must be set");
+        return ThreadPools.createFixedThreadPool(config.getNumThreads().get(),
+            "batch scanner " + config.getThreadName().get() + "-", false);
+      case BATCH_WRITER_BINNING_POOL:
+        return ThreadPools.createFixedThreadPool(1, "BinMutations", new SynchronousQueue<>(),
+            false);
+      case BATCH_WRITER_SEND_POOL:
+        Objects.requireNonNull(config.getNumThreads().get(), "Number of threads must be set");
+        return ThreadPools.createFixedThreadPool(config.getNumThreads().get(), "MutationWriter",
+            false);
+      case CONDITIONAL_WRITER_CLEANUP_TASK_POOL:
+        return ThreadPools.createFixedThreadPool(1, 3, TimeUnit.SECONDS,
+            "Conditional Writer Cleanup Thread", false);
+      case BLOOM_FILTER_LAYER_LOADER_POOL:
+        Objects.requireNonNull(config.getConfiguration().get(), "Configuration must be set");
+        return ThreadPools.createThreadPool(0,
+            new ConfigurationCopy(config.getConfiguration().get())
+                .getCount(Property.TSERV_BLOOM_LOAD_MAXCONCURRENT),
+            60, TimeUnit.SECONDS, "bloom-loader", false);
+      default:
+        throw new IllegalArgumentException("Unhandled thread pool usage value: " + usage.name());
+    }
+  }
+
+  public ScheduledThreadPoolExecutor newScheduledThreadPool(ScheduledThreadPoolType usage,
+      ThreadPoolConfig config) {
+    switch (usage) {
+      case SHARED_GENERAL_SCHEDULED_TASK_POOL:
+        Objects.requireNonNull(config.getConfiguration().get(), "Configuration must be set");
+        if (sharedScheduledThreadPool == null) {
+          sharedScheduledThreadPool = (ScheduledThreadPoolExecutor) ThreadPools
+              .createExecutorService(new ConfigurationCopy(config.getConfiguration().get()),
+                  Property.GENERAL_SIMPLETIMER_THREADPOOL_SIZE);
+        }
+        return sharedScheduledThreadPool;
```

Comment:
I don't see any sync here, so it seems like multiple instances of this thread pool could be created when this method is called by multiple threads.

Reply:
Resolved in 918b20f